### ISSUE TITLE
terminal path ordering not consistent on mac

### ISSUE THREAD

**Issue by dpastoor** [rstudio/rstudio#10997]
### System details

    RStudio Edition : Desktop
    RStudio Version : "Spotted Wakerobin" Daily (1b34f229, 2022-04-15) for macOS
    OS Version      : Mac
    R Version       : 4.1

### Steps to reproduce the problem

Open Rstudio via finder or other non-terminal launcher. Open a terminal and check paths

### Describe the problem in detail

There is a discrepancy in the path ordering in the RStudio terminal compared to using the terminal normally. Of particular interest is making sure there is consistency in /homebrew/bin

In rstudio terminal we see:

![image](https://user-images.githubusercontent.com/3196313/163594682-6ad9efae-3640-4a2a-b384-8ae1c8bfe10c.png)

regular terminal for default shell (fish) and in bash to be consistent with what rstudio launches

![image](https://user-images.githubusercontent.com/3196313/163594798-035a0618-0237-4d93-86af-f72bc403af08.png)

confirming bash is default shell in rstudio terminal
```
devins-mbp-2:quartoprojembed devin$ ps -p $$
  PID TTY           TIME CMD
24694 ttys007    0:00.01 bash -l
```

### Describe the behavior you expected

expected consistency between path implementations, if possible.

- [x] I have read the guide for [submitting good bug reports](https://github.com/rstudio/rstudio/wiki/Writing-Good-Bug-Reports).
- [x] I have installed the latest version of RStudio, and confirmed that the issue still persists.
- [] If I am reporting an RStudio crash, I have included a [diagnostics report](https://support.rstudio.com/hc/en-us/articles/200321257-Running-a-Diagnostics-Report).
- [x] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.

**Comment by kevinushey** [comment]
What is the output of `PATH= $SHELL -l -c 'printf "%s" "$PATH"'`, in your shell?

**Comment by kevinushey** [comment]
Also, do you see any relevant diagnostic entries in `~/.local/share/rstudio/log`?

**Comment by dpastoor** [comment]
sorry for the delay:

```
devins-mbp-2:environments.rstudio.com devin$ PATH= $SHELL -l -c 'printf "%s" "$PATH"'
/opt/homebrew/bin:/opt/homebrew/sbin:/Users/devin/go/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:.devins-mbp-2:environments.rstudio.com devin$ echo $PATH
/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/Users/devin/Applications/quarto/bin:/Applications/RStudio.app/Contents/MacOS

devins-mbp-2:environments.rstudio.com devin$ cat ~/.config/fish/config.fish 
source ~/.config/fish/env.fish
if status is-interactive
    # Commands to run in interactive sessions can go here
end
eval (/opt/homebrew/bin/brew shellenv)
starship init fish | source

fish_add_path ~/go/bin
```

In checking in `~/.local/share/rstudio/log` I didn't see anything relevant.

**Comment by dpastoor** [comment]
ah - I had downgraded to the preview release to revert some daily bugs, I need to rerun the above with the daily again, hold tight

**Comment by dpastoor** [comment]
kk 

```
RStudio 2022.06.0-daily+324 "Spotted Wakerobin" Daily (64ee43493bae9754916b9f18f91fe79a6ba9b8a9, 2022-04-25) for macOS
Mozilla/5.0 (Macintosh; Intel Mac OS X 12_1_0) AppleWebKit/537.36 (KHTML, like Gecko) QtWebEngine/5.12.10 Chrome/69.0.3497.128 Safari/537.36
```

```
devins-mbp-2:environments.rstudio.com devin$ PATH= $SHELL -l -c 'printf "%s" "$PATH"'
/opt/homebrew/bin:/opt/homebrew/sbin:/Users/devin/go/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:.
devins-mbp-2:environments.rstudio.com devin$ echo $PATH
/usr/bin:/usr/local/bin:/usr/bin:/bin:/usr/sbin:/sbin:/opt/X11/bin:/opt/homebrew/bin:/opt/homebrew/sbin:/Users/devin/go/bin:/Users/devin/Applications/quarto/bin:/Library/TeX/texbin:/usr/texbin:/Applications/RStudio.app/Contents/MacOS
```

**Comment by jgutman** [comment]
We somewhat recently (beginning of April) made some changes to fix the ordering of PATH entries on Mac in RStudio (both in R console and terminal), however based on the version numbers you've provided you are using all the latest versions with those fixes. However I'm not able to reproduce what you're seeing on RStudio Desktop 2022.6.0.374 with Mac OS 12.3.1 with default terminal of zsh. With "Enable Python integration" unchecked, my PATH is RStudio Terminal (zsh) is identical to my path in an external zsh terminal.

<img width="585" alt="image" src="https://user-images.githubusercontent.com/7817881/167473573-616164d4-4e54-4f6e-9c5e-ca672bfe7496.png">

<img width="901" alt="image" src="https://user-images.githubusercontent.com/7817881/167473698-749bdd5d-c14b-41d6-8b6c-f490f713b2fd.png">

<img width="1022" alt="image" src="https://user-images.githubusercontent.com/7817881/167473777-79d80e06-60f7-45ba-b0df-687be2fb4b9a.png">

Based on that I would expect your RStudio path to match what is being returned in your Bash shell. 

1) Does it make a difference if you check or uncheck the Enable Python integration option
2) Do you get the same path if you restart your regular terminal after switching your default shell to bash?
3) is there anything particular you're doing to add homebrew to your path that might make a difference? 

This may be a bit tricky to diagnose if we're not able to reproduce on other systems

**Comment by dpastoor** [comment]
@jgutman I'm using fish instead of zsh I'm not sure if they matters and I just followed the default instructions for fish around the homebrew env initializing. If you'd like I could ask around in sol-eng to see if others are running into this too?

**Comment by jgutman** [comment]
Yeah, sure, at this point we want to make sure that our earlier fix to Mac OS paths earlier in April hasn't made things more difficult for people.

I think the main thing would be comparing the behavior in what is set as your default shell for the RStudio terminal -- so if your RStudio terminal is set to use Bash by default, then you want to compare RStudio's terminal behavior to your external bash shell behavior

**Comment by jthomasmock** [comment]
Per @kevinushey - adding some notes here.

On Mac:

```
[1] "macOS Monterey 12.2.1"
R version 4.2.0 (2022-04-22)
platform       aarch64-apple-darwin20   
$mode
[1] "desktop"
$version
[1] '2022.6.0.421'
$long_version
[1] "2022.06.0-daily+421"
$release_name
[1] "Spotted Wakerobin"
```

I have the following command at the bottom of my `.profile` in the home directory.
```
export PATH="/Users/thomasmock/quarto-cli/package/dist/bin/:$PATH"
```

This adds a dev version of quarto to my `$PATH` and is then accessible by a terminal, integrated into RStudio or elsewhere.

However, RStudio is not "picking up" the `$PATH` addition as it is using an older version of Quarto with RStudio > Render, and the following commands point at a different executable. Calling `system("echo $PATH")` also returns a `$PATH` missing the export from above.

```
Sys.which("quarto")
#> quarto 
#> "/usr/local/bin/quarto" 
system("which quarto")
#> /usr/local/bin/quarto
```

**Comment by kevinushey** [comment]
@jthomasmock, can you also share the output of:

```
PATH= $SHELL -l -c 'printf "%s" "$PATH"'
```

**Comment by jthomasmock** [comment]
> @jthomasmock, can you also share the output of:
> 
> ```
> PATH= $SHELL -l -c 'printf "%s" "$PATH"'
> ```
> 
> as executed from a shell? The output of:
> 
> ```
> system("PATH= $SHELL -l -c 'printf \"%s\" \"$PATH\"'")
> ```
> 
> from the RStudio R console might also be useful.

Added to comment above so it's all in one place - thanks!

**Comment by mikebessuille** [comment]
Maybe related to https://github.com/rstudio/rstudio/issues/11822 ?
